homeassistant/components/synology_dsm/__init__.py

```diff
@@ -562,7 +643,22 @@ async def async_added_to_hass(self):
         self.async_on_remove(self._api.subscribe(self._api_key, self.unique_id))
 
 
-class SynologyDSMDeviceEntity(SynologyDSMEntity):
+class SynologyDSMCoordinatorEntity(SynologyDSMBaseEntity, CoordinatorEntity):
+    """Representation of a Synology NAS entry."""
+
+    def __init__(
+        self,
+        api: SynoApi,
+        entity_type: str,
+        entity_info: Dict[str, str],
+        coordinator: DataUpdateCoordinator,
+    ):
+        """Initialize the Synology DSM entity."""
+        super().__init__(api, entity_type, entity_info)
+        CoordinatorEntity.__init__(self, coordinator)
```

Comment:
This isn't recommended way to implement multiple inheritence. All the child classes should pass remaining non consumed parameters to their parents. That way it's enough to call `super().__init__` in each child once

Please refactor the class signatures of the entities to allow this. We may use unnamed arguments, `*args`, or keyword arguments if needed.